libres/lib/include/ert/res_util/memory.hpp

```diff
@@ -0,0 +1,59 @@
+/*
+   Copyright (C) 2011  Equinor ASA, Norway.
+
+   The file 'memory.hpp' is part of ERT - Ensemble based Reservoir Tool.
+
+   ERT is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   ERT is distributed in the hope that it will be useful, but WITHOUT ANY
+   WARRANTY; without even the implied warranty of MERCHANTABILITY or
+   FITNESS FOR A PARTICULAR PURPOSE.
+
+   See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+   for more details.
+*/
+
+#include <ert/python.hpp>
+
+#ifndef ERT_MEMORY_H
+#define ERT_MEMORY_H
+
+long SystemRamFree(void);
```

Comment:
As mentioned I think these functions should be tucked away in a implementation file - only indirectly accessible via a `struct meminfo`; if they are indeed exported in the header I think they should follow existing `snake_case` naming convention, maybe in a `memory` namespace:
```C++
namespace memory {
long system_free();
long process();
long process_max();
long process_max_rss();
}
```
or something similar.